datetime.asm

```diff
@@ -0,0 +1,411 @@
+    ; is not implemented yet)
+    
+    push hl
+        kld(hl, (current_year))
+        
+        ; subtract 1900 from it and put the result in a
+        ld de, -1900
+        add hl, de
+        ld a, l
+        
+        ; if a >= 100 we have year "20.."
+        cp 100
+        ld de, 0x0220
+        jr nc, .drawYear2000
+
+.drawYear1900:
+        ld de, 0x0a20
+        kcall(drawDecAPadded)
+        
+        ; "19"
+        ld a, '1'
+        ld de, 0x0220
+        pcall(drawChar)
+        ld a, '9'
+        ld de, 0x0620
+        pcall(drawChar)
+        
+        jr .endDrawYear
+
+.drawYear2000:
+        sub 100
+        ld de, 0x0a20
+        kcall(drawDecAPadded)
+        
+        ; "20"
+        ld a, '2'
+        ld de, 0x0220
+        pcall(drawChar)
+        ld a, '0'
+        ld de, 0x0620
+        pcall(drawChar)
+
+.endDrawYear:
+    pop hl
+    
+    ; now, draw the month and day
+    kld(a, (current_month))
+    inc a
+    ld de, 0x1620
+    kcall(drawDecAPadded)
+    kld(a, (current_day))
+    inc a
+    ld de, 0x2220
+    kcall(drawDecAPadded)
+    
+    ; the dashes
+    ld a, '-'
+    ld de, 0x1220
+    pcall(drawChar)
+    ld de, 0x1e20
+    pcall(drawChar)
+
+.drawTime:
+    kld(a, (current_hour))
+    ld de, 0x3220
+    kcall(drawDecAPadded)
+    kld(a, (current_minute))
+    ld de, 0x3c20
+    kcall(drawDecAPadded)
+    
+    ; the colon
+    ld a, ':'
+    ld de, 0x3a20
+    pcall(drawChar)
+    
+.drawSelection:
+    kld(hl, selected_field_indicator_x)
+    kld(a, (selected_field))
+    ld d, 0
+    ld e, a
+    add hl, de
+    ld d, (hl)
+    ld e, 0x1c
+    ld b, 3
+    kld(hl, caretUpIcon)
+    pcall(putSpriteOR)
+    ld e, 0x26
+    ld b, 3
+    kld(hl, caretDownIcon)
+    pcall(putSpriteOR)
+    
+    pcall(fastCopy)
+    
+.waitForKey:
+    pcall(flushKeys)
+    corelib(appWaitKey)
+    
+    ; arrow keys (move selected field)
+    cp kRight
+    jr nz, +_
+    kld(a, (selected_field))
+    inc a
+    cp 5
+    jr nz, $+3
+    ld a, 4
+    kld((selected_field), a)
+    kjp(.redraw)
+_:  
+    cp kLeft
+    jr nz, +_
+    kld(a, (selected_field))
+    dec a
+    cp -1
+    jr nz, $+3
+    ld a, 0
+    kld((selected_field), a)
+    kjp(.redraw)
+_:  
+    cp kUp
+    jr nz, +_
+    kcall(upPressed)
+    kjp(.redraw)
+_:  
+    cp kDown
+    jr nz, +_
+    kcall(downPressed)
+    kjp(.redraw)
+_:  
+    ; Clear (cancel)
+    cp kClear
+    jr nz, +_
+    ret
+_:  
+    ; Enter (save)
+    cp kEnter
+    jr nz, +_
+    kld(ix, (current_year))
+    kld((current_month), a)
+    ld l, a
+    kld((current_day), a)
+    ld h, a
+    kld((current_hour), a)
+    ld b, a
+    kld((current_minute), a)
+    ld c, a
+    ld d, 0
+    pcall(convertTimeToTicks)
+    pcall(setClock)
+    ret
+_:  
+    jr .waitForKey
+
+unsupported:
+    
+    kld(hl, clock_unsupported_message)
+    kld(de, clock_unsupported_options)
+    ld a, 0
+    ld b, 0
+    corelib(showMessage)
+    
+    ret
```

Comment:
I think it would be better to simply omit the clock option from the earlier screen if the clock is unsupported.